src/operation/watch/test.rs

```diff
@@ -0,0 +1,236 @@
+use std::time::Duration;
+
+use crate::{
+    bson::{doc, Document},
+    bson_util,
+    change_stream::ChangeStreamTarget,
+    cmap::{CommandResponse, StreamDescription},
+    concern::{ReadConcern, ReadConcernLevel},
+    operation::{Operation, Watch},
+    options::{
+        ChangeStreamOptions,
+        FullDocumentType,
+        ReadPreference,
+        SelectionCriteria,
+        StreamAddress,
+    },
+    Namespace,
+};
+
+fn build_test(
+    target: impl Into<ChangeStreamTarget>,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    mut expected_body: Document,
+) {
+    let watch = Watch::new(target.into(), pipeline, options).unwrap();
+
+    let mut cmd = watch.build(&StreamDescription::new_testing()).unwrap();
+
+    assert_eq!(cmd.name.as_str(), "aggregate");
+
+    bson_util::sort_document(&mut expected_body);
+    bson_util::sort_document(&mut cmd.body);
+
+    assert_eq!(cmd.body, expected_body);
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn build() {
+    let pipeline = vec![doc! { "$match": { "x": 3 } }];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .read_concern(ReadConcern::from(ReadConcernLevel::Available))
+        .max_await_time(Duration::from_millis(5))
+        .batch_size(10)
+        .full_document(FullDocumentType::UpdateLookup)
+        .build();
+
+    let expected_body = doc! {
```

Comment:
regarding my comment below, this `expected_body` should contain a `readConcern` field (since a read concern is specified above) but it doesn't.

Reply:
fixed